core/services/ardupilot_manager/mavlink_proxy/MAVLinkRouter.py

```diff
@@ -24,19 +24,21 @@ def _get_version(self) -> Optional[str]:
     def assemble_command(self, master: Endpoint) -> str:
         # Convert endpoint format to mavlink-router format
         def convert_endpoint(endpoint: Endpoint) -> str:
-            if endpoint.connection_type != EndpointType.TCP:
-                return str(endpoint)[str(endpoint).find(":") + 1 :]
             # TCP uses a special argument and only works with localhost
-            port = str(endpoint).split(":")[2]
-            return f"-t {port}"
+            if endpoint.connection_type == EndpointType.TCPServer:
+                port = str(endpoint).split(":")[2]
+                return f"-t {port}"
```

Comment:
Dejavu, this should use the full argument, always avoid single letter arguments.

Reply:
This is changed in the following commit. I just separated the changes because logically they are different changes.